collated/main/A0139655U.md

```diff
@@ -0,0 +1,1386 @@
+        assert args != null && regex != null;
+        String regexCopy = generateRegexForOneKeywordEscape(regex);
+        generateAndValidateMatcher(args, regexCopy);
+    }
+
+    /**
+     * Generates the matcher for the given args, where args has one keyword.
+     *
+     * @param args  user input of task to add.
+     * @param regex regex generated by generateStartOfRegex().
+     * @throws IllegalValueException  If args does not match the matcher.
+     */
+    private void generateMatcherForOneKeyword(String args, String regex) throws IllegalValueException {
+        assert args != null && regex != null;
+        String regexCopy = generateRegexForOneKeyword(regex);
+        generateAndValidateMatcher(args, regexCopy);
+    }
+    
+    /**
+     * Generates the regex to match an escaped string with one keyword 
+     * by concatenating the given regex generated by generateStartOfRegex() 
+     * with the regex for matching one keyword.
+     *
+     * @param regex regex generated by generateStartOfRegex().
+     * @return regex that matches an escaped string with one keyword.
+     */
+    private String generateRegexForOneKeywordEscape(String regex) {
+        assert regex != null;
+        return regex + REGEX_FIRST_DATE + REGEX_RECURRENCE_PRIORITY_CLOSE_BRACE;
+    }
+
+    /**
+     * Generates the regex to match an string with one keyword 
+     * by concatenating the given regex generated by generateStartOfRegex() 
+     * with the regex for matching one keyword.
+     *
+     * @param regex regex generated by generateStartOfRegex().
+     * @return regex that matches a string with one keyword.
+     */
+    private String generateRegexForOneKeyword(String regex) {
+        assert regex != null;
+        return regex + REGEX_CLOSE_BRACE + REGEX_FIRST_DATE + REGEX_RECURRENCE_PRIORITY_CLOSE_BRACE;
+    }
+    
+    /**
+     * Sets start date or end date depending on which matcher group was matched
+     *
+     * @param regex regex generated by generateStartOfRegex().
+     */
+    private void setStartOrEndDate(OptionalStringTask task, Matcher matcher) {
+        assert task != null && matcher != null;
+        task.startDate = validateStartDateFormatsOneToThree(matcher);
+        task.endDate = validateEndDateFormatsOneToThree(matcher);
+        assert task.startDate.isPresent() ^ task.endDate.isPresent();
+    }
+    
+    /**
+     * Reinitialises both start date and end date to be empty.
+     *
+     * @param task  object to store values for startDate and endDate.
+     */
+    private void reinitialiseStartAndEndDatesToEmpty(OptionalStringTask task) {
+        assert task != null;
+        task.startDate = Optional.empty();
+        task.endDate = Optional.empty();
+    }
+    
+    //TODO: HALP
+    /**
+     * Verifies if either startDate or endDate is present, and verifies whether 
+     * the date that is present is valid.
+     *
+     * @param startDate 
```

Comment:
you need to add the significance to the `@param` and `@return` statements, else remove these lines. 'cos in the current form, they don't provide any additional info